crates/node/builder/src/rpc.rs

```diff
@@ -310,6 +312,16 @@ pub struct RpcHandle<Node: FullNodeComponents, EthApi: EthApiTypes> {
     pub rpc_registry: RpcRegistry<Node, EthApi>,
 }
 
+impl<Node: FullNodeComponents, EthApi: EthApiTypes> RpcHandle<Node, EthApi> {
+    /// Returns an alloy provider from the rpc handle
+    pub fn new_alloy_provider(
+        &self,
+    ) -> Result<RootProvider<alloy_transport_http::Http<reqwest::Client>>, eyre::Error> {
+        let rpc_url = self.rpc_server_handles.rpc.http_url().unwrap().parse()?;
+        Ok(ProviderBuilder::new().on_http(rpc_url))
+    }
+}
```

Comment:
this looks like it could instead be a `http_provider` method directly on `RpcServerHandle`, similar to `http_client`
https://github.com/paradigmxyz/reth/blob/025885f2ad3fb881b1d2180e1998797f0b968b88/crates/rpc/rpc-builder/src/lib.rs#L2378-L2393